src/Specular/Internal/Incremental/PriorityQueue.js

```diff
@@ -0,0 +1,88 @@
+// new :: forall a.
+//   EffectFn4
+//     (Optional Any)                 -- Optional.none
+//     (Field a Mutable Int)          -- Priority
+//     (Field a Mutable Boolean)      -- Is the entry present in this queue?
+//     (Field a Mutable (Optional a)) -- Next entry with the same priority
+//   (PQ a)
+exports.new = function(none, priorityField, presentField, nextField) {
+  return {
+    none: none,
+    priorityField: priorityField,
+    presentField: presentField,
+    nextField: nextField,
+
+    priorityHeads: [],
```

Comment:
a list of pointers to the heads of the collection of items with the same priority? I yes, we can add it as a comment.

Reply:
Added comments, is it clear now?